tock/hours/admin.py

```diff
@@ -8,55 +8,53 @@
 
 
 class ReportingPeriodListFilter(admin.SimpleListFilter):
-  title = 'Reporting Period'
-  parameter_name = 'reporting_period'
+    title = 'Reporting Period'
+    parameter_name = 'reporting_period'
 
-  def lookups(self, request, model_admin):
-    reportingperiods = set([p.start_date
-                            for p in ReportingPeriod.objects.all()])
-    return [(p, p) for p in reportingperiods]
+    def lookups(self, request, model_admin):
+        reportingperiods = set(
```

Comment:
Nitpicky, but it's cheaper to get unique dates on `ReportingPeriod` from the database than to load all records, then filter in Python.